test/integration/cni/pod_egress_traffic_test.go

```diff
@@ -0,0 +1,286 @@
+		testerContainer coreV1.Container
+
+		// Primary node busybox deployment
+		primaryNodeDeployment *v1.Deployment
+		// Secondary node busybox deployment
+		secondaryNodeDeployment *v1.Deployment
+
+		primaryNodeIP   *string
+		secondaryNodeIP *string
+
+		isIPv4Cluster bool
+
+		// pods in running primary node
+		podsInPrimaryNode coreV1.PodList
+		// pods in running secondary node
+		podsInSecondaryNode coreV1.PodList
+		// IPs of PODs running in primary node
+		podsInPrimaryNodeIPs []string
+		// IPs of PODs running in secondary node
+		podsInSecondaryNodeIPs []string
+	)
+
+	BeforeEach(func() {
+		// initialize vars
+		err = nil
+		primaryNodeIP = nil
+		secondaryNodeIP = nil
+		testerContainer = coreV1.Container{}
+
+		// determine v4 or v6 cluster
+		clusterOutput, err := f.CloudServices.EKS().DescribeCluster(f.Options.ClusterName)
+		Expect(err).NotTo(HaveOccurred())
+		if *clusterOutput.Cluster.KubernetesNetworkConfig.IpFamily != "ipv6" {
+			isIPv4Cluster = true
+		}
+
+		// determine primary and secondary nodes IPs
+		PrimaryNodeInstance, err := f.CloudServices.EC2().DescribeInstance(k8sUtils.GetInstanceIDFromNode(primaryNode))
+		Expect(err).ToNot(HaveOccurred())
+		SecondaryNodeInstance, err := f.CloudServices.EC2().DescribeInstance(k8sUtils.GetInstanceIDFromNode(secondaryNode))
+		Expect(err).ToNot(HaveOccurred())
+		if isIPv4Cluster {
+			// Ipv6 address in node primary ENI is used for Ipv6 egress
+			primaryNodeIP = PrimaryNodeInstance.Ipv6Address
+			secondaryNodeIP = SecondaryNodeInstance.Ipv6Address
+		} else {
+			primaryNodeIP = PrimaryNodeInstance.PublicIpAddress
+			secondaryNodeIP = SecondaryNodeInstance.PublicIpAddress
+		}
+		Expect(primaryNodeIP).NotTo(BeNil())
+		Expect(secondaryNodeIP).NotTo(BeNil())
+
+		// initialize busybox container for testing later
+		testerContainer = manifest.NewCurlContainer().
+			Command([]string{"sleep", "3600"}).Build()
+
+		testerContainer.SecurityContext = &coreV1.SecurityContext{
+			RunAsUser:  aws.Int64(0),
+			Privileged: aws.Bool(true)} // ping needs privileged to run
+
+		// in IPv6 cluster, only need 2 pods for testing
+		var replicas = 2
+		if isIPv4Cluster {
+			// v6 cluster needs more pods to trigger secondary ENI
+			replicas = maxIPPerInterface * 2
+		}
+
+		By(fmt.Sprintf("creating test deployment on primary node: %s", primaryNode.Name))
+		primaryNodeDeployment = manifest.
+			NewDefaultDeploymentBuilder().
+			Container(testerContainer).
+			Replicas(replicas).
+			NodeName(primaryNode.Name).
+			PodLabel("node", "primary").
+			Name(fmt.Sprintf("primarynode-egress-tester")).
+			Build()
+		primaryNodeDeployment, err = f.K8sResourceManagers.
+			DeploymentManager().
+			CreateAndWaitTillDeploymentIsReady(primaryNodeDeployment, utils.DefaultDeploymentReadyTimeout)
+		Expect(err).ToNot(HaveOccurred())
+
+		if isIPv4Cluster {
+			// extra check for v4 cluster to assert pods are using both primary and secondary ENI
```

Comment:
nit: I don't think we need to check this explicitly. We can just set the replicas to `maxIPPerInterface * 2` regardless of v4 vs v6 and know that for v4, this will use multiple ENIs